clap_derive/src/derives/clap.rs

```diff
@@ -66,15 +66,26 @@ fn gen_for_struct(
     }
 }
 
-fn gen_for_enum(name: &Ident, attrs: &[Attribute], e: &DataEnum) -> proc_macro2::TokenStream {
+fn gen_for_enum(name: &Ident, attrs: &[Attribute], e: &DataEnum) -> TokenStream {
     let into_app = into_app::gen_for_enum(name);
     let from_arg_matches = from_argmatches::gen_for_enum(name);
     let subcommand = subcommand::gen_for_enum(name, attrs, e);
+
+    let arg_enum = if e.variants.iter().all(|v| match v.fields {
+        syn::Fields::Unit => true,
+        _ => false,
+    }) {
+        arg_enum::gen_for_enum(name, attrs, e)
+    } else {
+        quote!()
+    };
```

Comment:
This is pretty bad way to detect `ArgEnum`. For example, this is absolutely valid set of subcommands in structopt:
```
enum Sub {
    Foo,
    Bar,
}
```

I don't think this detection is even needed in the first place, see my comment about `#[derive(Clap)` above.

Reply:
Yes, it is valid set of commands and also valid set of arg enums which is detected here. If the variant has anything in it's field, then we can't implement `ArgEnum` on it.